File: DSA/Ayush_Mishra_590014844/Day_11_Question_3.py
def count_palindromic_substrings(s):
    count = 0
    for center in range(len(s)):
        for offset in (0, 1):
            l, r = center, center + offset
            while l >= 0 and r < len(s) and s[l] == s[r]:
                count += 1
                l -= 1
                r += 1
    return count

# Examples
print(count_palindromic_substrings("abc"))   # 3
print(count_palindromic_substrings("aaa"))   # 6

File: DSA/Ayush_Mishra_590014844/Day_11_Question_3.c
#include <stdio.h>
#include <string.h>
int count_palindromic_substrings(char *s) {
    int n = strlen(s), count = 0;

    for (int i = 0; i < n; i++) {
        for (int offset = 0; offset <= 1; offset++) {
            int l = i, r = i + offset;
            while (l >= 0 && r < n && s[l] == s[r]) {
                count++;
                l--;
                r++;
            }
        }
    }

    return count;
}

int main() {
    printf("%d\n", count_palindromic_substrings("abc"));  // 3
    printf("%d\n", count_palindromic_substrings("aaa"));  // 6
}
